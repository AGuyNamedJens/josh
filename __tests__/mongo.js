/* global describe expect it */
const Josh = require('../');
const Mongo = require('../providers/mongo/');
const config = require('./config-mongo.json');
/* config-mongo.json example contents
{
  "name": "josh-jest",
  "dbName": "testing",
  "user": "josh",
  "pass": "abcd1234",
  "host": "mydbidentifier.mlab.com",
  "port": 23456
}
*/

describe('MongoDB Initialization', () => {
  it('should initialize with credentials', () => {
    const provider = new Mongo(config);
    const josh = new Josh({ name: 'joshtest', provider });
    console.log(josh);
    josh.defer.then(() => {
      console.log(`Josh Initialize With ${josh.size} entries.`);
    });
    expect(josh).toMatchSnapshot();
  });
});
