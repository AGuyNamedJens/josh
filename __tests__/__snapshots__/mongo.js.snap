// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MongoDB Initialization should initialize with credentials 1`] = `Map {}`;
